pilot/pkg/model/context.go

```diff
@@ -508,7 +508,12 @@ type NodeMetadata struct {
 	Generator string `json:"GENERATOR,omitempty"`
 
 	// DNSCapture indicates whether the workload has enabled dns capture
-	DNSCapture string `json:"DNS_CAPTURE,omitempty"`
+	DNSCapture StringBool `json:"DNS_CAPTURE,omitempty"`
+
+	// DNSAutoAllocate indicates whether the workload should have auto allocated addresses for ServiceEntry
+	// This allows resolving ServiceEntries, which is especially useful for distinguishing TCP traffic
+	// This depends on DNSCapture.
+	DNSAutoAllocate StringBool `json:"DNS_AUTO_ALLOCATE,omitempty"`
```

Comment:
I think what we want is partial auto allocate, not all or none

Reply:
What do you want to auto allocate? I disagree we want any. Maybe we can add three options, but we must have a mode without autoallocation.

We need no-auto allocate, to minimize risk for VMs. Auto allocation is extremely tricky, as you found, and has countless edge cases we still need to iron out. However, we are planning to ship VM beta in 1.9, which requires a solid DNS story. This cannot be done with auto allocation enabled at this time

Reply:
Basically, I am not sure I agree at all that auto allocation should be partial, but even if it is, we still need to decouple it from DNS capture. If you want you can send a followup PR to make it configurable `{on, partial, off}`, but since I am not sure that is correct I don't want to do that in this PR.

Reply:
Sorry, I am late here and  not fully following -  What does partial auto allocation  mean here?

Reply:
@hzxuzhonghu was looking into making auto allocate only work for STATIC types since there is a bug where envoy's DNS requests will be captured as well. IMO, that is just a bug that needs to be fixed, but I haven't looked into the details yet.

Reply:
Got it. Thanks